### PR TITLE
[WIP] Fix windows tests issues and speed

### PR DESCRIPTION
**Pull request by sadikoff**
Trying to fix all tests issues. Still in progress...

### CODE REVIEW

**Comment by sadikoff** [comment]
Hm it looks like everything is ok, but I need several days to optimize some functions.
@weaverryan maybe you will have some time to review changes?

**Comment by weaverryan** [comment]
@sadikoff I still see some failing tests :/. Were they fixed, and then unfixed by later commits? And a few new tests are failing on Travis - probably we broke something by accident.

I would love to know what you think the cause is for the Windows tests failures :).

**Comment by sadikoff** [comment]
@weaverryan There was something broken, first time I started tests it takes 50 minutes to pass everything :)

And now It looks like some string replacements doesn't work as expected, All exceptions and failures were caused by wrong namespace or config used in cached env. Sometimes it was wrong from the beginning.

File: appveyor.yml [hunk 1]
```diff
@@ -26,8 +24,8 @@ init:
 environment:
     TEST_DATABASE_DSN: mysql://root:Password12!@127.0.0.1:3306/test_maker
     matrix:
-      - dependencies: lowest
-        php_ver_target: 7.1.3
+#      - dependencies: lowest
+#        php_ver_target: 7.1.3
```
Comment (weaverryan) [inline appveyor.yml]: I'm cool with this change - but let's delete it entirely :)

File: src/Test/MakerTestDetails.php [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
     private $deletedFiles = [];
 
-    private $filesToRevert = [];
+//    private $filesToRevert = [];
```
Comment (weaverryan) [inline src/Test/MakerTestDetails.php]: Delete this

File: src/Test/MakerTestDetails.php [hunk 3]
```diff
@@ -42,7 +42,9 @@
 
     private $commandAllowedToFail = false;
 
-    private $snapshotSuffix = '';
+//    private $snapshotSuffix = '';
```
Comment (weaverryan) [inline src/Test/MakerTestDetails.php]: Delete this

File: src/Test/MakerTestDetails.php [hunk 4]
```diff
@@ -138,17 +112,17 @@ public function getFilesToDelete(): array
...
+//    }
 
-    public function getFilesToRevert(): array
-    {
-        return $this->filesToRevert;
-    }
+//    public function getFilesToRevert(): array
+//    {
+//        return $this->filesToRevert;
+//    }
```
Comment (weaverryan) [inline src/Test/MakerTestDetails.php]: Remove this stuff :)

File: src/Test/MakerTestEnvironment.php [hunk 5]
```diff
@@ -72,6 +69,54 @@ public function getPath(): string
         return $this->path;
     }
 
+    private function changeRootNamespaceIfNeeded()
+    {
+        if ('App' !== ($rootNamespace = $this->testDetails->getRootNamespace())) {
```
Comment (weaverryan) [inline src/Test/MakerTestEnvironment.php]: Let's reverse this:

```php
if ('App' === ($rootNamespace = $this->testDetails->getRootNamespace())) {
    return;
}

// rest of code
```

File: src/Test/MakerTestEnvironment.php [hunk 6]
```diff
@@ -347,22 +340,14 @@ private function buildFlexSkeleton()
                 'replace' => '    <env name="SYMFONY_DEPRECATIONS_HELPER" value="weak_vendors" />'."\n".'    </php>',
             ],
         ];
+        $this->processReplacements($replacements, $this->flexPath);
+        // end of temp code
```
Comment (weaverryan) [inline src/Test/MakerTestEnvironment.php]: temp code? :)

File: tests/Maker/FunctionalTest.php [hunk 7]
```diff
@@ -85,7 +85,7 @@ public function getCommandTests()
             ])
             ->setFixtureFilesPath(__DIR__.'/../fixtures/MakeCommandInCustomRootNamespace')
             ->changeRootNamespace('Custom')
-            ->addPreMakeCommand('composer dump-autoload')
+//            ->addPreMakeCommand('composer dump-autoload')
```
Comment (weaverryan) [inline tests/Maker/FunctionalTest.php]: This can be removed now?

File: src/Test/MakerTestEnvironment.php (hunk 6 above)
Comment (sadikoff) [reply src/Test/MakerTestEnvironment.php]: i can remove this comment if you don't like it =) it is the end of code that we need to remove when deprecation notice goes away :)

**Comment by weaverryan** [comment]
@sadikoff Can I get one more favor? :) Can you rebase? This PR will merge cleanly, but rebasing is causing conflicts, and we need to squash the commits, unfortunately.

Thanks! I'm eager to merge this!

**Comment by weaverryan** [comment]
This is awesome. Finally got our Windows tests green. THANK you for your hard work on this @sadikoff & @nikophil!

**Pull request merged**